### PR TITLE
Sync to nodemaster at 7420835 (Oct 4 2016)

### PR DESCRIPTION
**Pull request by ianwjhalliday**
Sync to xplat to nodejs/master (2016-10-04) at 7420835390bcf44696349fad0498390959816acd

Update chakracore to Microsoft/ChakraCore@0b036046e7c7afa4dffcce9d78ac5f5154a9f0a1 with TypedArray constructor spec changes to fix various broken test cases.

Skip SharedArrayBuffer tests for node-chakracore.

Sync to more recent nodemaster to follow.

### CODE REVIEW

**Comment by ianwjhalliday** [comment]
Next sync will be to nodejs v7.1.x and come in shortly

File: test/message/timeout_throw.chakracore.out [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ReferenceError: 'undefined_reference_error_maker' is undefined
    at Anonymous function (*test*message*timeout_throw.js:*:*)
+   at ontimeout (timers.js:*:*)
```
Comment (kunalspathak) [inline test/message/timeout_throw.chakracore.out:3]: do you know why there is difference? I mean is this difference also in nodev8?

**Comment by kunalspathak** [comment]
> Next sync will be to nodejs v7.1.x

and not to nodejs/node's master branch?

**Comment by agarwal-sandeep** [comment]
commit 638860b title should be <= 50 characters

File: test/message/timeout_throw.chakracore.out (hunk 1 above)
Comment (agarwal-sandeep) [reply test/message/timeout_throw.chakracore.out:3]: Seems like we are now matching V8 but would be good to understand why was ontimeout function not in call stack before.

**Comment by agarwal-sandeep** [comment]
Rest looks good to me

**Comment by joaocgreis** [comment]
CI: https://ci.nodejs.org/view/All/job/chakracore-test-windows/25/

For some reason, it cannot find `ChakraCore.h`. Possibly something not being transferred from compile to test servers, I'll take a look.

**Comment by kunalspathak** [comment]
> For some reason, it cannot find ChakraCore.h.

This seemed to have failed [earlier](https://github.com/nodejs/node-chakracore/pull/139#issuecomment-261056308) as well.

File: test/message/timeout_throw.chakracore.out (hunk 1 above)
Comment (ianwjhalliday) [reply test/message/timeout_throw.chakracore.out:3]: It's just a refactoring in lib/timers.js.  A new timeout method was introduced before calling the user callback.

**Comment by ianwjhalliday** [comment]
@kunalspathak 
> > Next sync will be to nodejs v7.1.x

> and not to nodejs/node's master branch?

It will be v7.1.x from nodejs/node/master (e.g. nodejs/node@9314886053c30ad0623d891b3bf720446d2537f8)

**Comment by kunalspathak** [comment]
LGTM.

**Comment by ianwjhalliday** [comment]
@agarwal-sandeep 
> commit 638860b title should be <= 50 characters

I moved the long commit ref from the title to the body and rebased.

**Pull request merged**